src/Utils/DEM.jl

```diff
@@ -0,0 +1,77 @@
+
+
+abstract type DEM end
+
+############# DEM Interface begin ######################################
+
+"""
+    getCoordinate(dem::T,index::Tuple{Integer,Integer}) where T <: DEM
+
+    Get the coordinate for a certain index in the DEM. 
+    (All the DEM's are stored as a matrix of heights)
+"""
+function getCoordinate(dem::T,index::Tuple{Integer,Integer}) where T <: DEM
+     ErrorException("Not implemented for the following DEM type, $T")
+end
+
+
+"""
+    getIndex(dem::T,coordinate::Tuple{Real,Real}) where T <: DEM
+
+    Get the index of the DEM corresponding to the coordinate. 
+    (All the DEM's are stored as a matrix of heights)
+"""
+function getIndex(dem::T,coordinate::Tuple{Real,Real}) where T <: DEM
+    ErrorException("Not implemented for the following DEM type, $T")
+end
+
+############# DEM Interface end ######################################
+
+
+"""
+    TandemxDEM <: DEM
+
+    A DEM implementation for Tandem-X DEM data.
+"""
+struct TandemxDEM <: DEM
+    heights::Array{Union{Missing,Float32},2}
```

Comment:
Should it always med 'Missing'? Why do we want that?

Reply:
Okay, they are invalid pixels. Will there alway be invalid pixels, though. What if you create a subset over land?

Reply:
The function should work for all cases. I might change it to NaN. I think NaN is a normal part of the Float type and then I can avoid the union